### PR TITLE
Allow multiple apps with same repo

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

**Comment by Pespiri** [comment]
please merge with the latest changes in master

File: src/components/page-configuration/change-repository-form.tsx [hunk 1]
```diff
@@ -1,60 +1,91 @@
...
   useEffect(() => {
-    setRepository(props.repository);
-    setUpdateRepositoryProgress(false);
-    return () => setPreviousRepository(repository);
+    setEditedRepository(currentRepository);
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [props.repository]);
+  }, [currentRepository]);
 
   const handleSubmit = (ev) => {
```
Comment (Pespiri) [inline src/components/page-configuration/change-repository-form.tsx:58]: add typing (remember to update imports `FormEvent from 'react'`)
```suggestion
  const handleSubmit = (ev: FormEvent<HTMLFormElement>): void => {
```

File: src/components/page-configuration/change-repository-form.tsx [hunk 2]
```diff
@@ -1,60 +1,91 @@
...
+  const [currentRepository, setCurrentRepository] = useState(props.repository);
+  const [editedRepository, setEditedRepository] = useState(props.repository);
+  const [useAcknowledgeWarnings, setAcknowledgeWarnings] = useState(false);
+  const [modifyState, saveFunc, resetState] = useSaveRepository(props.appName);
+  const applicationRegistration =
+    modifyState.data?.applicationRegistration ?? undefined;
+  const operationWarnings = modifyState.data?.warnings ?? undefined;
   const [updateRepositoryProgress, setUpdateRepositoryProgress] =
     useState(false);
+  const webhookURL = `https://webhook.${radixZoneDNS}/events/github?appName=${props.appName}`;
```
Comment (Pespiri) [inline src/components/page-configuration/change-repository-form.tsx:51]: sort definitions for readability
```suggestion
  const [currentRepository, setCurrentRepository] = useState(props.repository);
  const [editedRepository, setEditedRepository] = useState(props.repository);
  const [useAcknowledgeWarnings, setAcknowledgeWarnings] = useState(false);
  const [modifyState, saveFunc, resetState] = useSaveRepository(props.appName);
  const [updateRepositoryProgress, setUpdateRepositoryProgress] =
    useState(false);

  const webhookURL = `https://webhook.${radixZoneDNS}/events/github?appName=${props.appName}`;
  const applicationRegistration = modifyState.data?.applicationRegistration;
  const operationWarnings = modifyState.data?.warnings;
```

File: src/components/page-configuration/change-repository-form.tsx [hunk 3]
```diff
@@ -1,60 +1,91 @@
...
+  ApplicationRegistrationModelValidationMap,
+} from '../../models/application-registration';
+import imageDeployKey from '../configure-application-github/deploy-key02.png';
+import imageWebhook from '../configure-application-github/webhook01.png';
+import { Code } from '../code';
 import { copyToClipboard } from '../../utils/string';
+import { configVariables } from '../../utils/config';
+import { useSaveRepository } from './use-save-repository';
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (Pespiri) [inline src/components/page-configuration/change-repository-form.tsx:29]: remove unnecessary lint-ignore
```suggestion
```

File: src/components/page-configuration/change-repository-form.tsx [hunk 4]
```diff
@@ -1,60 +1,91 @@
...
+export interface ChangeRepositoryFormProps {
+  appName: string;
+  repository: string;
+  acknowledgeWarnings?: boolean;
+  app?: ApplicationRegistrationModel;
+}
+
+export const ChangeRepositoryForm = (
+  props: ChangeRepositoryFormProps
+): JSX.Element => {
```
Comment (Pespiri) [inline src/components/page-configuration/change-repository-form.tsx:41]: optional ease of use cleanup: spread input props
```jsx
export const ChangeRepositoryForm = ({
  appName,
  repository,
  acknowledgeWarnings,
  app,
}: ChangeRepositoryFormProps): JSX.Element => {
```

**Review by Pespiri — CHANGES_REQUESTED** [review]

File: src/components/page-configuration/change-repository-form.tsx [hunk 5]
```diff
@@ -1,60 +1,91 @@
...
+import {
+  ApplicationRegistrationModel,
+  ApplicationRegistrationModelValidationMap,
+} from '../../models/application-registration';
+import imageDeployKey from '../configure-application-github/deploy-key02.png';
+import imageWebhook from '../configure-application-github/webhook01.png';
+import { Code } from '../code';
 import { copyToClipboard } from '../../utils/string';
+import { configVariables } from '../../utils/config';
+import { useSaveRepository } from './use-save-repository';
```
[the remark this answers is not in the document]
Comment (satr) [reply src/components/page-configuration/change-repository-form.tsx:27]: Fixed

File: src/components/page-configuration/change-repository-form.tsx (hunk 4 above)
Comment (satr) [reply src/components/page-configuration/change-repository-form.tsx]: Fixed

File: src/components/page-configuration/change-repository-form.tsx [hunk 6]
```diff
@@ -1,60 +1,94 @@
...
+  const webhookURL = `https://webhook.${radixZoneDNS}/events/github?appName=${appName}`;
+  const applicationRegistration = modifyState.data?.applicationRegistration;
+  const operationWarnings = modifyState.data?.warnings;
+
   useEffect(() => {
-    setRepository(props.repository);
-    setUpdateRepositoryProgress(false);
-    return () => setPreviousRepository(repository);
+    setEditedRepository(currentRepository);
     // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Pespiri) [inline src/components/page-configuration/change-repository-form.tsx:56]: remove unnecessary eslint ignore rule
```suggestion
```

**Review by Pespiri — APPROVED** [review]

File: src/components/page-configuration/change-repository-form.tsx [hunk 7]
```diff
@@ -1,60 +1,91 @@
...
+      applicationRegistration &&
+      !operationWarnings
+    ) {
+      setCurrentRepository(applicationRegistration.repository);
+    }
+  }, [modifyState, applicationRegistration, operationWarnings]);
+
   return (
     <Accordion className="accordion" chevronPosition="right">
       <Accordion.Item>
```
[the remark this answers is not in the document]
Comment (satr) [reply src/components/page-configuration/change-repository-form.tsx:93]: Fixed

File: src/components/page-configuration/change-repository-form.tsx (hunk 6 above)
Comment (satr) [reply src/components/page-configuration/change-repository-form.tsx]: Fixed

**Review by Pespiri — APPROVED** [review]

**Pull request merged**